### PR TITLE
Must have

### PR DESCRIPTION
**Pull request by saharin88**
Please see changes

Pull Request for Issue # .

### CODE REVIEW

File: libraries/src/Language/Text.php [hunk 1]
```diff
@@ -378,6 +378,16 @@ public static function script($string = null, $jsSafe = false, $interpretBackSla
...
 			);
 		}
+		
+		if(is_array($string))
+		{
+			foreach ($string as $key)
+			{
+				self::script($key, $jsSafe, $interpretBackSlashes);
+			}
+
```
Comment (Quy) [inline libraries/src/Language/Text.php:388]: Remove tabs.

**Comment by ghost** [comment]
@saharin88 can you please describe what this Pull Request is about?

File: libraries/src/Language/Text.php (hunk 1 above)
Comment (saharin88) [reply libraries/src/Language/Text.php:388]: I use Joomla code style for PhpStorm

File: libraries/src/Language/Text.php [hunk 2]
```diff
@@ -378,6 +378,16 @@ public static function script($string = null, $jsSafe = false, $interpretBackSla
 				'deprecated'
 			);
 		}
+
```
Comment (Quy) [inline libraries/src/Language/Text.php:381]:
```suggestion


```

File: libraries/src/Language/Text.php [hunk 3]
```diff
@@ -378,6 +378,16 @@ public static function script($string = null, $jsSafe = false, $interpretBackSla
 				'deprecated'
 			);
 		}
+		
+		if(is_array($string))
```
Comment (Quy) [inline libraries/src/Language/Text.php:382]:
```suggestion
		if (is_array($string))
```

**Comment by Bakual** [comment]
Looks like you want to be able to pass an array to Text::script(). What would the reason be for that?
If you have multiple strings to add, why not just do the foreach in your code? Should be simple and probably more performant.

**Comment by saharin88** [comment]
> Looks like you want to be able to pass an array to Text::script(). What would the reason be for that?
> If you have multiple strings to add, why not just do the foreach in your code? Should be simple and probably more performant.

I think better 
Text::script(['key1','key2','key3']);
than so
Text::script('key1');
Text::script('key2');
Text::script('key3');

File: libraries/src/Language/Text.php (hunk 1 above)
Comment (Quy) [inline libraries/src/Language/Text.php:388]:
```suggestion


```

**Comment by Bakual** [comment]
Personally, I don't think it's better. I'd just do this
````
$keys = array('key1','key2','key3');
foreach ($keys as $key) {
    Text::script($key);
}
````

**Comment by saharin88** [comment]
> Personally, I don't think it's better. I'd just do this

but I do not think so
we can leave it as it is

**Comment by mbabker** [comment]
There is no practical benefit here.  What this PR does is basically exactly what @Bakual is demonstrating, but within the method's code (so it is recursively calling itself).  If it were optimized to natively support arrays without a recursive/loop call, maybe there would be a benefit in taking this, otherwise I don't see the point.

**Comment by ghost** [comment]
should this be closed?

**Comment by ghost** [comment]
@HLeithner is it a problem if i close a PR and you wan't to merge it later?

**Comment by HLeithner** [comment]
closed as annotated there is no benefit, adding a loop to line 403 makes more sense but have to build for j4.

**Pull request closed without merging**